### PR TITLE
Feat/dataprotection-backup-schedule -- Initial Client, Models & TF Schema

### PR DESCRIPTION
**Pull request by GilTeraSky**
1. **What this PR does / why we need it**:
       This PR includes the Swagger API Models & Clients along with the Terraform Schema Design for creating a scheduled 
       backup on a cluster

2. **Which issue(s) this PR fixes**
        -

3. **Additional information**
       -

4. **Special notes for your reviewer**
       -

### CODE REVIEW

File: internal/resources/cluster/backupschedule/resource_schema.go [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+	meKey         = "key"
+	meOperatorKey = "operator"
+	meValuesKey   = "values"
+)
+
+var backupScheduleResourceSchema = map[string]*schema.Schema{
+	nameKey:        nameSchema,
+	clusterNameKey: clusterNameSchema,
+	specKey:        specSchema,
+	common.MetaKey: common.Meta,
```
Comment (rnarenpujari) [inline internal/resources/cluster/backupschedule/resource_schema.go:74]: I see the data protection resource in [this](https://github.com/vmware/terraform-provider-tanzu-mission-control/pull/265) PR also has `managementClusterNameKey` and `provisionerNameKey`. Is that not needed here?

File: internal/models/cluster/backupschedule/exec.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+//	return &value
+//}
+//
+//// Pointer returns a pointer to a freshly-allocated VmwareTanzuManageV1alpha1ClusterDataprotectionBackupHookErrorMode.
+//func (m VmwareTanzuManageV1alpha1ClusterDataprotectionBackupHookErrorMode) Pointer() *VmwareTanzuManageV1alpha1ClusterDataprotectionBackupHookErrorMode {
+//	return &m
+//}
+//
+//const (
+//
```
Comment (rnarenpujari) [inline internal/models/cluster/backupschedule/exec.go:64]: looks like you've moved this stuff to a different file. Guess this can be dropped then?

**Review by rnarenpujari — COMMENTED** [review]
LGTM

File: internal/resources/cluster/backupschedule/resource_schema.go (hunk 1 above)
Comment (GilTeraSky) [reply internal/resources/cluster/backupschedule/resource_schema.go:74]: I am not sure if it is required in the API itself, I will finish testing today and adjust

File: internal/models/cluster/backupschedule/exec.go (hunk 2 above)
Comment (GilTeraSky) [reply internal/models/cluster/backupschedule/exec.go]: removed.

File: internal/resources/cluster/backupschedule/resource_schema.go (hunk 1 above)
Comment (GilTeraSky) [reply internal/resources/cluster/backupschedule/resource_schema.go]: All fixed

**Review by rnarenpujari — APPROVED** [review]

**Pull request merged**